src/solvers/flattening/bv_utils.cpp

```diff
@@ -488,20 +488,28 @@ bvt bv_utilst::shift(const bvt &src, const shiftt s, std::size_t dist)
 
     switch(s)
     {
-    case shiftt::LEFT:
+    case shiftt::SHIFT_LEFT:
       l=(dist<=i?src[i-dist]:const_literal(false));
       break;
 
-    case shiftt::ARIGHT:
+    case shiftt::SHIFT_ARIGHT:
       l=(i+dist<src.size()?src[i+dist]:src[src.size()-1]); // sign bit
       break;
 
-    case shiftt::LRIGHT:
+    case shiftt::SHIFT_LRIGHT:
       l=(i+dist<src.size()?src[i+dist]:const_literal(false));
       break;
 
+    case shiftt::ROTATE_LEFT:
+      l=(dist<=i?src[i-dist]:src[src.size()-dist+i]);
+      break;
+
+    case shiftt::ROTATE_RIGHT:
+      l=(i+dist<src.size()?src[i+dist]:src[i+dist-src.size()]);
```

Comment:
Shouldn't this be `(i+dist)%src.size()`?

Reply:
That would be equivalent.